### PR TITLE
Add new sync providers

### PR DESCRIPTION
**Pull request by six7**

### CODE REVIEW

File: pages/sync/tokens-studio.en.mdx [hunk 1]
```diff
@@ -0,0 +1,57 @@
+# Tokens Studio Platform sync setup guide
+Tokens Studio has a standalone web-based platform for dynamic creation and management of design decisions, currently in Beta! 
+
+→ More information on the Studio Platform and sign up for Beta access here. TODO
```
Comment (SamIam4Hyma) [inline pages/sync/tokens-studio.en.mdx:4]: @six7 reminder to add the URL hyperlink where it says `here` and remove the TODO in the doc.

**Review by SamIam4Hyma — COMMENTED** [review]
content added, there's one TODO to add the link to studio.

File: pages/sync/bitbucket.en.mdx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+You can use the Tokens Studio plugin native integration with Bitbucket to sync your design tokens to a repository of your choice. 
+
+We support bi-directional sync, meaning you can use the plugin to: 
+- `push` JSON files full of design tokens to Bitbucket
+- `pull` the tokens stored in Bitbucket into any Figma file
+
+This means the design tokens living in code are the source of truth for our design decisions, which can be shared between design and development teams. 
+
+This doc outlines how to set up a Bitbucket repository and add it as a `Sync provider` in the plugin.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:12]:
```suggestion
This document outlines how to set up a Bitbucket repository and add it as a `Sync provider` in the plugin. 
```

File: pages/sync/bitbucket.en.mdx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+We support bi-directional sync, meaning you can use the plugin to: 
+- `push` JSON files full of design tokens to Bitbucket
+- `pull` the tokens stored in Bitbucket into any Figma file
+
+This means the design tokens living in code are the source of truth for our design decisions, which can be shared between design and development teams. 
+
+This doc outlines how to set up a Bitbucket repository and add it as a `Sync provider` in the plugin. 
+## How it works
+- Set up a `repository` and `personal access token` in Bitbucket.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:14]:
```suggestion
- Set up a `repository` and an `app password` in Bitbucket.
```

File: pages/sync/bitbucket.en.mdx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+### 1. Create a new repository
+Sign into your account and select the `Create` button, and select `Repository`.
+- Choose a specific and memorable `repository name` which will be used in the plugin as the destination for your design tokens. 
+- Review the options, [learn more about those in the Bitbucket docs.](https://support.atlassian.com/bitbucket-cloud/docs/create-a-git-repository/)
+	- `Add a README file` needs to be checked
+		- This is mandatory because the plugin can not sync to an empty repository.
+	- The other options are up to you.
+
+You are now looking at your new repository! Well done! 
+- `Save the URL of the repository` somewhere for safekeeping and plugin configuration later.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:30]:
```suggestion
```

This feels strange, its not like we dont see this anymore later. Let's remove.

File: pages/sync/bitbucket.en.mdx [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+- Select `Create app passwords`
+- Add a `Name` of what the app password is for
+	- Example: `test-token repo sync to tokens studio`
+- Select the `permissions` for app passwords
+	- `Account` = `Read`
+	- `Repositories` = `Write`
+- Select the `Create` button
+- The page will display the `New app password dialog`
+
+ `Copy the generated token` and store it securely, as you will need it to configure the connection with the Tokens Studio plugin.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:45]:
```suggestion
**Copy the generated token** and store it securely, as you will need it to configure the connection with the Tokens Studio plugin.
```

Can we find some consistency in what we mark with backticks? Usually I do this for stuff that has a specifi meaning, here it seems its just to emphasize. Let's use bold for stuff like this instead.

File: pages/sync/bitbucket.en.mdx [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+- `Copy your Bitbucket username` alongside the app password, as it's required to authenticate your Bitbucket account.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:46]:
```suggestion
- **Copy your Bitbucket username** alongside the app password, as it's required to authenticate your Bitbucket account. 
```

File: pages/sync/bitbucket.en.mdx [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+- Example: `radix ui components`
+
+#### Bitbucket username
+The `username` of your Bitbucket account is used as a part of the authentication process between the plugin and Bitbucket.
+
+#### App password 
+The `App password` you created and saved from the steps above. 
+
+#### Repository (owner/repo)
+When you created the `repository` from the steps above there was a step to `save the URL of the repository somewhere safe`.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:68]:
```suggestion
When you created the `repository` from the steps above you chose a certain name. It's now time to fill this in.
```

File: pages/sync/bitbucket.en.mdx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+The `App password` you created and saved from the steps above. 
+
+#### Repository (owner/repo)
+When you created the `repository` from the steps above there was a step to `save the URL of the repository somewhere safe`. 
+- The owner/repository details for this input it in the URL after the `bitbucket.org/` 
+- For example, if my URL says `https://butbucket.org/amazingdesigner/radixtokens` I am going to enter:
+	- `amazingdesigner/radixtokens` 
+
+#### Branch 
+Your engineers might tell you what `respository branch` to add here, but if you followed the steps above exactly, your repository was created with `main` as the"primary branch".
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:74]: missing space

```suggestion
Your engineers might tell you what `respository branch` to add here, but if you followed the steps above exactly, your repository was created with `main` as the "primary branch".
```

File: pages/sync/bitbucket.en.mdx [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+- So you will enter `main`
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:75]:
```suggestion
- So you will enter `main` (unless you need a different branch specified by your engineers)
```

File: pages/sync/bitbucket.en.mdx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+	- If we enter `tokens.json` 
+		- a file called "tokens.json" would be created in our Bitbucket repository
+		- all tokens we create in the plugin will be synced in this one file. 
+
+##### Folder Path
+- `folder` would have each token set in its own JSON file saved to this folder. 
+	- If we enter `tokens` 
+		- a folder called "tokens" would be created in our Bitbucket repository
+		- each token set we create in the plugin would live in that folder. 
+	- This "multi-file sync" option is a `pro` feature and requires a paid licence to Tokens Studio.
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:93]:
```suggestion
	- This "multi-file sync" option is a `Pro` feature and requires a paid licence to Tokens Studio. 
```

File: pages/sync/bitbucket.en.mdx [hunk 11]
```diff
@@ -0,0 +1,99 @@
...
+
+##### Folder Path
+- `folder` would have each token set in its own JSON file saved to this folder. 
+	- If we enter `tokens` 
+		- a folder called "tokens" would be created in our Bitbucket repository
+		- each token set we create in the plugin would live in that folder. 
+	- This "multi-file sync" option is a `pro` feature and requires a paid licence to Tokens Studio. 
+	- If you are using the `Themes feature`, and your engineers will transform your tokens with Style Dictionary to use in code, you want this option. 
+
+This setting also has some impact on teams using Tokens Studio where not all team members are on the pro licence or when you want a "read-only" option for certain team members
```
Comment (six7) [inline pages/sync/bitbucket.en.mdx:96]:
```suggestion
This setting also has some impact on teams using Tokens Studio where not all team members are on the Pro licence. If someone does not have a Pro licence, they will only be able to **read** from your tokens, not **write** changes back.
```

File: pages/sync/tokens-studio.en.mdx (hunk 1 above)
Comment (six7) [reply pages/sync/tokens-studio.en.mdx:4]:
```suggestion
→ More information on the Studio Platform and sign up for Beta access [here](https://tokens.studio/studio).
```

File: pages/sync/tokens-studio.en.mdx [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+## How it works
+Set up a `project` and `API key` in Tokens Studio
+- `Configure Tokens Studio as a sync provider` within the plugin.
+- Seamlessly `sync your design tokens` between Figma and Tokens Studio to maintain consistency across design and development workflows.
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:9]:
```suggestion
- Set up a `project` in Tokens Studio
- For your organization, create an API Key in Tokens Studio
- Configure Tokens Studio as a sync provider within the plugin.
- Seamlessly `sync your design tokens` between Figma and Tokens Studio to maintain consistency across design and development workflows.

```

File: pages/sync/tokens-studio.en.mdx [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+## Tokens studio sync setup Instructions
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:10]:
```suggestion
## Tokens Studio sync setup Instructions
```

File: pages/sync/tokens-studio.en.mdx [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+If you haven't already, sign up for an account at [Tokens Studio](https://app.tokens.studio/)
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:11]:
```suggestion
If you haven't already, sign up for an account at [Tokens Studio](https://app.tokens.studio/). Note that sign-ups will be waitlisted, so you'll need to be invited to the Beta first.
```

File: pages/sync/tokens-studio.en.mdx [hunk 15]
```diff
@@ -0,0 +1,57 @@
...
+### 1. Create a new project
+Once you've logged into your Tokens Studio account:
+- Select `New Project` from the top-right corner of the home page
+	- Give your project a title.
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:15]:
```suggestion
- Give your project a title.
```

File: pages/sync/tokens-studio.en.mdx [hunk 16]
```diff
@@ -0,0 +1,57 @@
...
+- Now you are looking at the dashboard of your new project. 
+- Locate the `Connection String` (ID of the project) on the right side of the page
+- Use the `copy button` to copy the connection string to your clipboard
+- `Save the Connection ID` and 
+	- `Store it somewhere safe` as it's needed for the plugin configuration.
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:20]:
```suggestion
- `Store it somewhere safe` as it's needed for the plugin configuration. 
```

File: pages/sync/tokens-studio.en.mdx [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+- Select `API Keys`
+- Select `Create API Key`
+- Add a `Name` of what the API Key is for. 
+	- Example: `fimga-sync`
+	- Option to add a `Description` for the API Key
+		- Example: `test-token sync to plugin`
+- Under `Group` select `Power User`
+- Select `create key` 
+- `Copy the generated API key` and 
+	- `Store it somewhere safe` as it's needed for the plugin configuration.
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:34]:
```suggestion
- Add a `Name` of what the API Key is for. 
  - Example: `figma-sync`
 	- Optional: add a description for the API key
- Under `Group` select `Power User` or `Admin` (or Read-only if that's all you need)
- Select `Create key` 
- Copy the generated API key and store it for later
```

File: pages/sync/tokens-studio.en.mdx [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+
+#### API Key
+The `API Key` you created and saved from the steps above. 
+
+#### ID (connection string)
+Enter the `Connection String ID` you copied and saved from the steps above. 
+- This can typically be found in the project section of your Tokens Stdio sync dashboard.
+
+### 4. Save and do the initial push
+
```
Comment (six7) [inline pages/sync/tokens-studio.en.mdx:57]:
```suggestion
### 4. Save credentials

Initially you will pull the tokens that you have stored in your project.

You can now go and either create tokens inside the plugin, or on the platform.
- If you make changes on the platform there's a button in the bottom left of the plugin allowing you to pull the latest values.
- If you make changes in the plugin, they will automatically be pushed to Tokens Studio on save.

Note that dynamic token sets (sets created as a graph) will be read-only in the plugin.
```

**Review by six7 — COMMENTED** [review]
Couple of comments with suggestions @SamIam4Hyma

File: pages/sync/bitbucket.en.mdx (hunk 3 above)
Comment (SamIam4Hyma) [reply pages/sync/bitbucket.en.mdx:14]: Thank you! This was a copy paste error. good catch

File: pages/sync/bitbucket.en.mdx (hunk 4 above)
Comment (SamIam4Hyma) [reply pages/sync/bitbucket.en.mdx:30]: This is the same content we provide on all other docs. Designers don't know they need to keep a copy of the URL so they can use parts of it in the form later. We can remove, but if there are problems we can add back later.

File: pages/sync/bitbucket.en.mdx (hunk 5 above)
Comment (SamIam4Hyma) [reply pages/sync/bitbucket.en.mdx:45]: Yes! Ideally this happens with the Editor. This isn't my strength and Ajit is aware and ready to help with these consistency checks.

**Comment by SamIam4Hyma** [comment]
Suggestion all committed, will monitor feedback and questions from community to see if any of the suggestions need to be reverted. (:

**Pull request merged**